From 5e7b14608b0870de0bdd74cd1c18c039e267e415 Mon Sep 17 00:00:00 2001
From: Kevin Brace <kevinbrace@gmx.com>
Date: Wed, 6 Jul 2016 19:30:16 -0500
Subject: [PATCH 02/21] viaIGAInitCommon will now initialize certain legacy VGA
 registers

During ACPI S3 State resume functionality testing, Sylvania g netbook
(VIA NanoBook reference design equivalent) was losing analog VGA output
when it resumes from standby. Not initializing legacy VGA's miscellaneous
output register appears to cause this bug.

Signed-off-by: Kevin Brace <kevinbrace@gmx.com>
---
 src/via_display.c | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/src/via_display.c b/src/via_display.c
index 1ee8934..0dcb659 100644
--- a/src/via_display.c
+++ b/src/via_display.c
@@ -288,6 +288,23 @@ viaIGAInitCommon(ScrnInfoPtr pScrn)
      *             1: Enable */
     ViaSeqMask(hwp, 0x10, 0x01, 0x01);
 
+    temp = hwp->readEnable(hwp);
+    DEBUG(xf86DrvMsg(pScrn->scrnIndex, X_INFO,
+                        "Enable Register: 0x%02X\n", temp));
+    hwp->writeEnable(hwp, temp | 0x01);
+
+    temp = hwp->readMiscOut(hwp);
+    DEBUG(xf86DrvMsg(pScrn->scrnIndex, X_INFO,
+                        "Misc. Register: 0x%02X\n", temp));
+    hwp->writeMiscOut(hwp, temp | 0x2E);
+
+    temp = hwp->readEnable(hwp);
+    DEBUG(xf86DrvMsg(pScrn->scrnIndex, X_INFO,
+                        "Enable Register: 0x%02X\n", temp));
+    temp = hwp->readMiscOut(hwp);
+    DEBUG(xf86DrvMsg(pScrn->scrnIndex, X_INFO,
+                        "Misc. Register: 0x%02X\n", temp));
+
     ViaCRTCSetGraphicsRegisters(pScrn);
     ViaCRTCSetAttributeRegisters(pScrn);
 
-- 
1.9.1

